File: KITE-Example-Test/src/main/java/org/webrtc/kite/example/pages/GoogleResultPage.java
package org.webrtc.kite.example.pages;

import io.cosmosoftware.kite.exception.KiteInteractionException;
import io.cosmosoftware.kite.pages.BasePage;
import io.cosmosoftware.kite.report.KiteLogger;
import io.cosmosoftware.kite.interfaces.Runner;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;


public class GoogleResultPage extends BasePage {
  
  @FindBy(className = "LC20lb")
  WebElement result;
  
  public GoogleResultPage(Runner runner) {
    super(runner);
  }
  
  public String getTitle() {
    return webDriver.getTitle();
  }
  
  public void openFirstResult() throws KiteInteractionException {
    waitUntilVisibilityOf(result, 10);
    result.click();
  }
}

File: KITE-Example-Test/js/pages/GoogleResultPage.js
const {By, until} = require('selenium-webdriver');

const elements = {
  result: By.className('LC20lb'),
}

class GoogleResultPage {
  constructor(driver) {
    this.driver = driver;
  }

  async openFirstResult() {
    let result = await this.driver.wait(until.elementLocated(elements.result));
    await result.click();
  }

  async getTitle() {
    return await this.driver.getTitle();
  }

}
module.exports = GoogleResultPage